### PR TITLE
(GH-66) Fix go to definition

### LINKED ISSUE #66: Go to defintion does not work with paths with leading slash
Part of `go to definition` feature is navigating to the source file where the function is defined if it's not in the current file. On platforms with paths that have leading slashes, this does not work.

On ubuntu 18.04, vscode 1.28, and pdk 1.7.1 type a simple `file` declaration and watch the log:

```
D, [2018-10-24T16:11:25.404160 #34999] DEBUG -- : --- INBOUND
{"jsonrpc":"2.0","id":9,"method":"textDocument/definition","params":{"textDocument":{"uri":"file:///home/james/Documents/foo.pp"},"position":{"line":0,"character":2}}}
---
D, [2018-10-24T16:11:25.418013 #34999] DEBUG -- : --- OUTBOUND
{"jsonrpc":"2.0","id":9,"result":[{"uri":"file:////opt/puppetlabs/pdk/share/cache/ruby/2.4.0/gems/puppet-6.0.2/lib/puppet/type/file.rb","range":{"start":{"line":14,"character":0},"end":{"line":14,"character":1024}}}]}
---
D, [2018-10-24T16:11:25.554092 #34999] DEBUG -- : --- INBOUND
{"jsonrpc":"2.0","id":10,"method":"textDocument/hover","params":{"textDocument":{"uri":"file:///home/james/Documents/foo.pp"},"position":{"line":0,"character":5}}}
---
```

Note the file uri differences between the definition request and the hover request. It's not a pathing issue in detecting pdk folders because the file is present at that location.

The lines below are where we are building the file uris:

https://github.com/lingua-pupuli/puppet-editor-services/blob/113a7e292db42b07e208233ca4c1b932fa312b7b/lib/puppet-languageserver/manifest/definition_provider.rb#L73

https://github.com/lingua-pupuli/puppet-editor-services/blob/113a7e292db42b07e208233ca4c1b932fa312b7b/lib/puppet-languageserver/manifest/definition_provider.rb#L88

### PR DESCRIPTION
**Pull request by jpogran**
Fixes #66, reported in https://github.com/lingua-pupuli/puppet-vscode/issues/331

### CODE REVIEW

File: lib/puppet-languageserver/manifest/definition_provider.rb [hunk 1]
```diff
@@ -85,14 +85,19 @@ def self.function_name(func_name)
...
+          'uri' => build_uri(item.source),
           'fromline' => item.line,
           'fromchar' => 0,
           'toline' => item.line,
           'tochar' => 1024
         )
       end
       private_class_method :function_name
+
+      def self.build_uri(path)
```
Comment (glennsarti) [inline lib/puppet-languageserver/manifest/definition_provider.rb]: I wonder if this should be a generic helper. Although, nothing else is broken yet...

Comment (glennsarti) [inline lib/puppet-languageserver/manifest/definition_provider.rb]: Do we need unit tests on this?

Comment (jpogran) [reply lib/puppet-languageserver/manifest/definition_provider.rb]: I see this being useful in other providers that handle opening or doing things to files

Comment (jpogran) [reply lib/puppet-languageserver/manifest/definition_provider.rb]: Sure

**Comment by glennsarti** [comment]
@jpogran #70 will pin rubocop.

**Comment by glennsarti** [comment]
I'm happy to merge this without red CI.  Rubocop is tested last for a reason :-)

**Pull request merged**